peachjam/models/core_document_model.py

```diff
@@ -79,6 +81,54 @@ def update_languages(self):
             self.languages = langs
             self.save()
 
+    def update_extracted_citations(self):
+        """Update the current work's ExtractedCitations."""
+        related_works = Work.objects.filter(
+            frbr_uri__in=self.fetch_related_works_frbr_uris()
+        )
+
+        # delete existing extracted citations
+        ExtractedCitation.objects.filter(citing_work=self).delete()
+
+        for related_work in related_works:
+            ExtractedCitation.objects.get_or_create(
+                citing_work=self, target_work=related_work
+            )
+
+    def fetch_related_works_frbr_uris(self):
+        """Returns a list of unique work_frbr_uris,
+        taken from CitationLink objects(for PDFs) and all <a href="/akn/..."> embedded HTML links."""
+        for document in self.documents.all():
+
+            work_frbr_uris = []
+
+            if document.content_html:
+                root = html.fromstring(document.content_html)
+                for a in root.xpath('//a[starts-with(@href, "/akn")]'):
```

Comment:
Both this link or the citation link below could be expression frbr uris. For example:

`/akn/za/act/2009/1/eng@2011-01-01`

We need to track citations between works, not expressions. The work URI for the above is:

`/akn/za/act/2009/1`

So in both cases, I think we need to parse the URI and then get the work component:

```
try:
  work_frbr_uris.add(FrbrUri.parse(a.attrib["href"]).work_uri())
except ValueError:
  # ignore malformed FRBR URIs
  pass
```

Reply:
The `FrbrUri` object is from `cobalt.uri`